File: frontend/src/api/incident.ts
import api from './client'

export interface IncidentOrder {
  id: number
  order_no: string
  elder_id: number
  risk_event_id?: number
  incident_type: string
  severity: string
  incident_date: string
  incident_time: string
  location?: string
  impact_scope: string
  responsibility: string
  responsible_person?: string
  handling_result: string
  preventive_measures?: string
  description: string
  immediate_actions?: string
  medical_treatment?: string
  family_notified: string
  family_notification_time?: string
  family_response?: string
  reported_by_id?: number
  handled_by?: string
  reviewed_by?: string
  status: string
  closure_date?: string
  remark?: string
  created_at: string
  updated_at: string
}

export const incidentApi = {
  getList: (params?: {
    page?: number
    page_size?: number
    incident_type?: string
    severity?: string
    status?: string
  }) => {
    return api.get<any>('/incident-orders', { params })
  },

  getDetail: (id: number) => {
    return api.get<IncidentOrder>(`/incident-orders/${id}`)
  },

  create: (data: Partial<IncidentOrder> & { elder_id: number }) => {
    return api.post<IncidentOrder>('/incident-orders', data)
  },

  update: (id: number, data: Partial<IncidentOrder>) => {
    return api.put<IncidentOrder>(`/incident-orders/${id}`, data)
  },

  updateStatus: (id: number, status: string) => {
    return api.patch<IncidentOrder>(`/incident-orders/${id}/status`, { status })
  },

  remove: (id: number) => {
    return api.delete(`/incident-orders/${id}`)
  },
}

File: backend/app/schemas/incident.py
from datetime import date, datetime, time
from pydantic import BaseModel
from typing import Optional


class IncidentOrderBase(BaseModel):
    incident_type: str
    severity: str = "serious"
    incident_date: date
    incident_time: time
    location: Optional[str] = None
    impact_scope: str
    responsibility: str
    responsible_person: Optional[str] = None
    handling_result: str
    preventive_measures: Optional[str] = None
    description: str
    immediate_actions: Optional[str] = None
    medical_treatment: Optional[str] = None
    family_notified: str = "unknown"
    family_response: Optional[str] = None
    handled_by: Optional[str] = None
    reviewed_by: Optional[str] = None
    status: str = "pending"
    closure_date: Optional[date] = None
    remark: Optional[str] = None


class IncidentOrderCreate(IncidentOrderBase):
    elder_id: int
    risk_event_id: Optional[int] = None


class IncidentOrderUpdate(BaseModel):
    incident_type: Optional[str] = None
    severity: Optional[str] = None
    incident_date: Optional[date] = None
    incident_time: Optional[time] = None
    location: Optional[str] = None
    impact_scope: Optional[str] = None
    responsibility: Optional[str] = None
    responsible_person: Optional[str] = None
    handling_result: Optional[str] = None
    preventive_measures: Optional[str] = None
    description: Optional[str] = None
    immediate_actions: Optional[str] = None
    medical_treatment: Optional[str] = None
    family_notified: Optional[str] = None
    family_response: Optional[str] = None
    handled_by: Optional[str] = None
    reviewed_by: Optional[str] = None
    status: Optional[str] = None
    closure_date: Optional[date] = None
    remark: Optional[str] = None


class IncidentOrderResponse(IncidentOrderBase):
    id: int
    order_no: str
    elder_id: int
    risk_event_id: Optional[int] = None
    reported_by_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
